src/core/dynamic_batch_scheduler.cc

```diff
@@ -198,60 +155,44 @@ DynamicBatchScheduler::Enqueue(std::unique_ptr<InferenceRequest>& request)
     // Assuming no error is returned, this call takes ownership of
     // 'request' and so we can't use it after this point.
     RETURN_IF_ERROR(queue_.Enqueue(request->Priority(), request));
-
-    // If there are any idle runners and the queued batch size is greater or
-    // equal to next preferred batch size, then wake one up to service this
-    // request. We do the actual wake outside of the lock to avoid having the
-    // woken thread immediately block on the lock
-    wake_runner = (idle_scheduler_thread_cnt_ > 0);
-
-    // We may wake up runner less often if we don't enforce equal shape within
-    // a batch, otherwise must always wake up runner to check it
-    if (enforce_equal_shape_tensors_.empty()) {
-      wake_runner &= (queued_batch_size_ >= next_preferred_batch_size_);
-    }
   }
 
-  if (wake_runner) {
-    cv_.notify_one();
-  }
+  cv_.notify_one();
 
   return Status::Success;
 }
 
 void
 DynamicBatchScheduler::SchedulerThread(
-    const uint32_t runner_id, const int nice,
-    const std::shared_ptr<std::atomic<bool>>& rthread_exit,
-    std::promise<bool>* is_initialized)
+    TritonModel* model, const int nice, std::promise<bool>* is_initialized)
 {
 #ifndef _WIN32
   if (setpriority(PRIO_PROCESS, syscall(SYS_gettid), nice) == 0) {
-    LOG_VERBOSE(1) << "Starting dynamic-batch scheduler thread " << runner_id
-                   << " at nice " << nice << "...";
+    LOG_VERBOSE(1) << "Starting dynamic-batch scheduler thread for "
+                   << model->Name() << " at nice " << nice << "...";
   } else {
-    LOG_VERBOSE(1) << "Starting dynamic-batch scheduler thread " << runner_id
-                   << " at default nice (requested nice " << nice
-                   << " failed)...";
+    LOG_VERBOSE(1) << "Starting dynamic-batch scheduler thread for "
+                   << model->Name() << " at default nice (requested nice "
+                   << nice << " failed)...";
   }
 #else
-  LOG_VERBOSE(1) << "Starting dynamic-batch scheduler thread " << runner_id
-                 << " at default nice...";
+  LOG_VERBOSE(1) << "Starting dynamic-batch scheduler thread for "
+                 << model->Name() << " at default nice...";
 #endif
 
   // Initialize using the thread. If error then just exit this thread
   // now... that means the corresponding model instance will not have
   // any runner and so will not get used for execution.
-  Status startup_status = OnInit_(runner_id);
+  Status startup_status = model->Initialize();
```

Comment:
This initialize and warmup are ok for now, but given the new reorg we are moving towards they don't belong here. When a TritonModel object is created it can initialize and warmup itself, and then create the scheduler for itself.

Reply:
I agree... I will keep note of this in my next PR for handling multiple instances.